layer/main.py

```diff
@@ -665,11 +668,26 @@ def _check_latest_version() -> None:
     current_version = pkg_resources.get_distribution("layer").version
     if current_version != latest_version:
         print(
-            "You are using the version {current_version} but the latest version is {latest_version}, please upgrade with 'pip install --upgrade layer'"
+            f"You are using the version {current_version} but the latest version is {latest_version}, please upgrade with 'pip install --upgrade layer'"
         )
     set_has_shown_update_message(True)
 
 
+def _check_python_version() -> None:
+    if has_shown_python_version_message():
+        return
+
+    import platform
+
+    major, minor, _ = platform.python_version_tuple()
+
+    if major != "3" or minor not in ["7", "8"]:
```

Comment:
What about minor 9 and 10 ?

Reply:
I think we only support Python 3.7 and 3.8? cc @mecevit

Reply:
@eminn you should know this better, because of cloudpickle limitations, i believe we only 3.7 + 3.8 images for our executors, no?